packages/test/src/workflows/unhandled-rejection.ts

```diff
@@ -0,0 +1,23 @@
+import { proxyActivities } from '@temporalio/workflow';
+import type * as activities from '../activities';
+
+const { echo } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '1 minute',
+});
+
+export async function throwUnhandledRejection({ crashWorker }: { crashWorker: boolean }): Promise<void> {
+  const p1 = (async () => {
+    await echo('a');
+  })();
+
+  const p2 = (async () => {
+    if (crashWorker) {
+      throw 'not an error to crash the worker';
+    } else {
+      throw new Error('unhandled rejection');
+    }
+  })();
+
+  await p1;
+  await p2;
```

Comment:
Does this not just rethrow here? Before this PR, is this treated any different than if I just replaced this line with `throw new Error('whatever')`?

Reply:
The problem is that the second async function throws without anything catching it.
The way v8 deals with it is by providing a hook for unhandled rejections: https://v8.github.io/api/head/classv8_1_1Isolate.html#a702f0ba4e5dee8a98aeb92239d58784e.
This is exposed in node with `process.on('unhandledRejection')`.

Reply:
But since that promise is `await`ed on here, it throws here right? If you removed `await p2` I think it'd be unhandled, but the promise is explicitly awaited on.